src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVVariantCallerInternal.java

```diff
@@ -0,0 +1,338 @@
+        return new ArrayList<>(Arrays.asList(Allele.create(new String(reference.getReferenceBases(null, new SimpleInterval(contig, start, start)).getBases()), true), Allele.create("<INV>")));
+    }
+
+    @VisibleForTesting
+    static String produceVariantId(final BreakpointAllele breakpointAllele) {
+        final BreakpointAlleleInversion inversionAllele = (BreakpointAllele.BreakpointAlleleInversion) breakpointAllele;
+        return inversionAllele.getInversionType().name() + SVConstants.VARIANT_ID_FIELD_SEPARATER +
+                inversionAllele.leftAlignedLeftBreakpoint.getContig() + SVConstants.VARIANT_ID_FIELD_SEPARATER +
+                inversionAllele.leftAlignedLeftBreakpoint.getStart() + SVConstants.VARIANT_ID_FIELD_SEPARATER +
+                inversionAllele.leftAlignedRightBreakpoint.getStart();
+    }
+
+    /**
+     * Auxiliary struct for extracting information from {@link BreakpointAllele.BreakpointAlleleInversion} to make the call.
+     * todo: add documentation, especially which concept this aux is helping with
+     */
+    static final class AuxStruct implements Serializable {
+        private static final long serialVersionUID = 1L;
+
+        // 5 entries: (min mapping qualities of the two alignment regions that formed the chimeric alignment)
+        //            (min alignment length of the two alignment regions that formed the chimeric alignment)
+        //            (local assembly id)
+        //            (contig id)
+        //            (mapping information (collected from each chimeric alignment) of the inserted sequence)
+        // reason for going with this way is such that the order will not be affected by how Spark distribute assemblies over executors
+        final List<Tuple5<Integer, Integer, String, String, List<String>>> breakpointAlignmentInformation;
+//        // following lists have the same length as input list to {@link collectInfo()}
+//        final List<Integer> mqs;                // mapping qualities
+//        final List<Integer> minAlignLengths;    // alignment length of the flanking regions of each identified breakpoint
+//        final List<String> asmIds;              // local assembly ids
+//        final List<String> assembledContigIds;  // contig ids
+
+//        final List<String> insertionMappings;   // mapping information (collected from each chimeric alignment) of the inserted sequence
+
+        final int numAssembledBreakpoints;            // todo: number of identified breakpoints for this contig
+        final int highMqMappings;                     // number of breakpoints with high quality mappings, i.e. those with flanking mapping regions of a mapQ higher than a threshold
+        final int maxMinAlignLength;                  // maximum alignment length of the flanking regions out of all identified breakpoints, i.e. max of {@code minAlignLengths}
+
+        AuxStruct(final List<Tuple3<String, String, ChimericAlignment>> assembledBreakpoints) {
```

Comment:
Based on our conversation, please either replace this class with a map with named fields or a specifically named class (`BreakpointAlignmentInformation` or something similar) that has properly named fields (as opposed to the `Tuple5` structure or something similar that you're using above).